src/main/java/edu/unc/lib/boxc/migration/cdm/services/GroupMappingService.java

```diff
@@ -197,6 +185,14 @@ private void ensureMappingState(GroupMappingOptions options) {
         }
     }
 
+    private String buildKey(List<String> values) {
+        String joinedValues = null;
+        if (!values.isEmpty()) {
```

Comment:
I don't think this will catch if all the entries in the list are empty, just if the list itself is empty. It's possible that it could result in entries with all blanks for the group field would produce the key ",," (for a 3 field grouping request), which might result in everything with blank fields getting grouped together (although I would have thought the tests would have caught that?).  One way to check if they are all empty is `values.stream().allMatch(String::isEmpty)`